lib/screens/creategame.dart

```diff
@@ -1,189 +1,488 @@
+        break;
+    }
+  }
+
+  _joinRoomListener(Map message) {
+    switch (message["action"]) {
+      case "set_id":
+        // Set the player ID.
+        playerId = message["data"]["playerId"];
+        Map joinDetails = {
+          "roomId": joinRoomId,
+          "name": user.displayName,
+          "playerId": playerId
+        };
+        game.send("join_game", json.encode(joinDetails));
+        break;
+
+      ///
+      /// Each time a new player joins, we need to
+      ///   * record the new list of players
+      ///   * rebuild the list of all the players
+      ///
+      case "joined":
+        setState(() {
+          isLoading = false;
+        });
+        if (playersList.length == 0) {
+          playersList = (message["data"])["players"];
+          // print(playersList.toString());
+          currPlayer = new Player(name: user.displayName);
+          // Assign the ID of the player
+          currPlayer.id = playerId;
+        }
+        final players = Provider.of<PlayerList>(context, listen: false);
+        players.addCurrPlayer(currPlayer);
+        players.removeAll();
+        List<Player> lp = [];
+        for (var player in (message["data"])["players"]) {
+          Player p;
+          if ((message["data"])["lobbyLeader"]["id"] == player["id"])
+            p = new Player(
+                name: player["name"], id: player["id"], lobbyLeader: true);
+          else
+            p = new Player(name: player["name"], id: player["id"]);
+          lp.add(p);
+        }
+        players.addPlayers(lp);
+        // force rebuild
+        // game.removeListener(_joinRoomListener);
+        Navigator.pushReplacement(
+            context,
+            new MaterialPageRoute(
+              builder: (BuildContext context) => WaitingPage(
+                roomId: message["data"]["roomId"],
+              ),
+            ));
+        break;
+      case "roomisfull":
+        showCustomDialogWithImage(context, "full");
+        game.disconnect();
+        setState(() {
+          isLoading = false;
+        });
+        break;
+      case "invalid room":
+        showCustomDialogWithImage(context, "invalid");
+        game.disconnect();
+        setState(() {
+          isLoading = false;
+        });
+        break;
+    }
+  }
+
+  /// -----------------------------------------
+  /// Sends a create game signal to the server
+  /// -----------------------------------------
+  _onCreateGame() {
+    // Send a message to server to create a new game
+    // and then move to join room page.
+
+    ///
+    /// Ask to be notified when messages related to the game
+    /// are sent by the server, also creates the connection.
+    ///
+
+    game.connect();
+    game.addListener(_createGameListener);
+    setState(() {
+      isLoading = true;
+    });
+    // Forces a rebuild
+    // setState(() {});
+  }
+
+  _onJoinGame() {
+    // Connect to the
+    // socket.
+    game.addListener(_joinRoomListener);
+    game.connect();
```

Comment:
Find out why this creates a game, rather than joining the one provided after "set-id"